utils/zerovec/src/zerovec/mod.rs

```diff
@@ -392,6 +392,65 @@ where
         self.as_slice().iter().map(T::from_unaligned)
     }
 
+    /// Mutates each element according to a given function, meant to be
+    /// a more convenient version of [`ZeroVec::iter_mut()`] which serves fewer use cases.
+    ///
+    /// This will convert the ZeroVec into an owned ZeroVec if not already the case.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    /// use zerovec::ule::AsULE;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let mut zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// zerovec.for_each_mut(|item| *item += 1);
+    ///
+    /// assert_eq!(zerovec.to_vec(), &[212, 282, 422, 462]);
```

Comment:
Nit: Here and below:

```suggestion
    /// assert_eq!(zerovec.to_vec(), &[212, 282, 422, 462]);
    /// assert!(matches!(ZeroVec::Owned(_), zerovec));
```